packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -681,10 +683,16 @@ export class Pipeline extends PipelineBase {
   private generateNameForDefaultBucketKeyAlias(): string {
     const prefix = 'alias/codepipeline-';
     const maxAliasLength = 256;
-    const uniqueId = Names.uniqueId(this);
-    // take the last 256 - (prefix length) characters of uniqueId
-    const startIndex = Math.max(0, uniqueId.length - (maxAliasLength - prefix.length));
-    return prefix + uniqueId.substring(startIndex).toLowerCase();
+    const maxUniqueIdLength = maxAliasLength - prefix.length;
+    if (FeatureFlags.of(this).isEnabled(cxapi.CODEPIPELINE_CROSS_ACCOUNT_KEY_ALIAS_STACK_SAFE_UNIQUE_ID)) {
+      const uniqueId = Names.stackSafeUniqueId(this, maxUniqueIdLength);
+      return prefix + uniqueId.toLowerCase();
+    } else {
+      const uniqueId = Names.uniqueId(this);
+      // take the last 256 - (prefix length) characters of uniqueId
+      const startIndex = Math.max(0, uniqueId.length - (maxUniqueIdLength));
+      return prefix + uniqueId.substring(startIndex).toLowerCase();
```

Comment:
We're repeating `return prefix + <sth>.toLowerCase();` in both branches of the `if`. Maybe extract a variable, something like `suffix: string;`, in the block above, then set it to different values in the `if`-`else` above, and then put `return prefix + suffix.toLowerCase();` after the `if`, as the last statement of the method?

Reply:
Updated. Looks much better now.